crates/accelerate/src/two_qubit_decompose.rs

```diff
@@ -1023,17 +1048,18 @@ impl TwoQubitWeylDecomposition {
     #[pyo3(signature = (euler_basis=None, simplify=false, atol=None))]
     fn circuit(
         &self,
+        py: Python,
         euler_basis: Option<PyBackedStr>,
         simplify: bool,
         atol: Option<f64>,
-    ) -> PyResult<TwoQubitGateSequence> {
+    ) -> PyResult<CircuitData> {
         let euler_basis: EulerBasis = match euler_basis {
             Some(basis) => EulerBasis::__new__(basis.deref())?,
             None => self.default_euler_basis,
         };
         let target_1q_basis_list: Vec<EulerBasis> = vec![euler_basis];
 
-        let mut gate_sequence = Vec::new();
+        let mut gate_sequence: WeylCircuitSequence = Vec::with_capacity(21);
```

Comment:
Why is the capacity specifically 21?

Reply:
An arbitrary 2q decomposition is 3 CX plus local operations, which with single-rotation Eulers is up to 3 gates per local per qubit.  Naively, this can run to 27 gates (3x CX interspersed into 4x 6 1q local operations), but maybe there's a way through the decomposition that guarantees that the last set of 6 is never needed?  I don't have the maths off the top of my head.

Reply:
I counted it out as 21 for the two qubit decomposer: https://github.com/Qiskit/qiskit/blob/main/crates/accelerate/src/two_qubit_decompose.rs#L1836-L1842 (but might have miscounted) and was just reusing that number here.